### PR TITLE
Fixed failing sqlalchemy inventory.

### PR DESCRIPTION
**Pull request by potiuk**
This time sqlalchemy causes docs build failures

---
**^ Add meaningful description above**

Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/master/CONTRIBUTING.rst#pull-request-guidelines)** for more information.
In case of fundamental code change, Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvements+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in [UPDATING.md](https://github.com/apache/airflow/blob/master/UPDATING.md).

### CODE REVIEW

**Comment by potiuk** [comment]
@mik-laj  - this really calls for some action - see #14989 .

File: docs/exts/docs_build/third_party_inventories.py [hunk 1]
```diff
@@ -24,7 +24,7 @@
     'pandas': 'https://pandas.pydata.org/pandas-docs/stable',
     'python': 'https://docs.python.org/3',
     'requests': 'https://requests.readthedocs.io/en/master',
-    'sqlalchemy': 'https://docs.sqlalchemy.org/en/latest',
+    'sqlalchemy': 'https://docs.sqlalchemy.org/en/stable',
```
Comment (kaxil) [inline docs/exts/docs_build/third_party_inventories.py:27]: This does not exists either !

**Review by kaxil — CHANGES_REQUESTED** [review]
I don't think the issue was with sqlalchemy, the retry and timeout was because of pymongo which also got fixed, since we fetch them in pool, it showed different error in CI.

https://github.com/apache/airflow/pull/14986 -- this was fixed here (rebasing on Master should help). I can retrieve https://docs.sqlalchemy.org/en/latest/objects.inv

**Comment by potiuk** [comment]
> I don't think the issue was with sqlalchemy, the retry and timeout was because of pymongo which also got fixed, since we fetch them in pool, it showed different error in CI.
> 
> #14986 -- this was fixed here (rebasing on Master should help). I can retrieve https://docs.sqlalchemy.org/en/latest/objects.inv

No. it was sqlalchemy. I checked it and got infite recursion. Simply the problem was fixed in the meantime

**Comment by kaxil** [comment]
> > I don't think the issue was with sqlalchemy, the retry and timeout was because of pymongo which also got fixed, since we fetch them in pool, it showed different error in CI.
> > #14986 -- this was fixed here (rebasing on Master should help). I can retrieve https://docs.sqlalchemy.org/en/latest/objects.inv
> 
> No. it was sqlalchemy. I checked it and got infite recursion. Simply the problem was fixed in the meantime

aah -- interesting -- yeah we need to make the doc build fail-safe in this case. thanks for creating the issue #14989

**Pull request closed without merging**